torchtune/models/clip/_text_encoder.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+from typing import Optional
+
+import torch
+from torch import nn, Tensor
+
+from torchtune.modules.attention_utils import _MaskType
+
+
+class CLIPTextEncoder(nn.Module):
+    """
+    Text encoder for CLIP.
```

Comment:
mind briefly describing the architecture? probably a normal transformer, but uses a different MLP activation?

Reply:
it's just a normal transformer. the MLP activation is still GELU, it's just a faster and less-precise version of it (tho not actually faster these days, this is a relic of the ancient year of 2021)